relentless/core.py

```diff
@@ -149,7 +149,7 @@ def __init__(self, types):
         self._data = {}
         for i in self.types:
             for j in self.types:
-                if j >= i:
+                if str(j) >= str(i):
```

Comment:
This will force all types into strings, which we don't necessarily want to do. Let's revert it back, and just check that an error is raised if two types are mixed that cannot be compared.